src/components/pill/pill.style.js

```diff
@@ -28,6 +28,11 @@ const PillStyle = styled.span`
     const { colors } = baseTheme;
     const correctedTheme = setTheme(theme);
     const styleSet = styleConfig(correctedTheme)[pillRole];
+    const { boxShadow, hoverColor } = styleConfig(correctedTheme);
```

Comment:
As `correctedTheme` isn't used anywhere else, might it be worth instead of that, doing something like this on line 29:
`const correctedThemeConfig = styleConfig(setTheme(theme));`

Then you could use `correctedThemeConfig` rather than repeating the `styleConfig(correctedTheme)` call